app/controllers/buildings_controller.rb

```diff
@@ -0,0 +1,27 @@
+class BuildingsController < ApplicationController
+  def create
+    # should always create buildings with a portfolio
+    @portfolio = Portfolio.find(params[:portfolio_id])
+    @building = @portfolio.buildings.create(building_params)
```

Comment:
For now, this is ok.
I'll be going over how to render our data as json so that we can request it from the front end. When we do, you have to not only render a json object with the data in it, but you also have to propagate any errors encountered when creating. Right now, you don't have a condition to catch if the building isn't successfully created i.e.

`portfolio = Portfolio.find(params[:portfolio_id]`
`building = portfolio.buildings.new(building_params)`
`if building.save`
     `render success json object`
`else`
    `render error json object`